### PR TITLE
Feature data model

### PR DESCRIPTION
**Pull request by Tchilo**
- Create users table
- Create likes table
- Create posts table
- Create comments table 
- Add user references to posts, comments, and likes
- Add post references to comments and likes

### CODE REVIEW

File: db/migrate/20220321234229_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class CreateUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :photo
+      t.text :bio
+
```
Comment (MahmoudBakr23) [inline db/migrate/20220321234229_create_users.rb:8]: - [x] Well done so far building the migrations of your app and adding the references correctly. However, its seems that you might have missed adding the `posts_counter` column, so let's do that. :pray: :smile:

File: db/migrate/20220321235014_create_posts.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+
+      t.timestamps
```
Comment (MahmoudBakr23) [inline db/migrate/20220321235014_create_posts.rb:9]: - [x] The same thing here, please add the `comments_counter` and the `likes_counter` columns. :+1:

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @Tchilo  😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done creating the migrations and adding the references between the tables. 

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️

- [x] Kindly fix the MIT link for a completely professional repo. 

_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220321234229_create_users.rb (hunk 1 above)
Comment (Tchilo) [reply db/migrate/20220321234229_create_users.rb:8]: thanks 😸

File: db/migrate/20220321235014_create_posts.rb (hunk 2 above)
Comment (Tchilo) [reply db/migrate/20220321235014_create_posts.rb:9]: thanks 👍🏽 I forgot haha

**Review by addod19 — APPROVED** [review]
### Hi @Tchilo, 

Your project is complete!  Nice job adding the comments and likes counters :+1:  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**